### PR TITLE
chore: Update code for reuse v5.0.2

### PR DESCRIPTION
**Pull request by klmcadams**
- Reordered code to be in chronological order
- Fix how years are being checked & updated in files
- Create separate functions to add and update the headers (which are called in the recursive & non-recursive file checks)
- Check if file contains reuse information. If not, add the header to the file. Otherwise, update the header

### CODE REVIEW

**Comment by RobPasMue** [comment]
I'll have a look at this asap, but probably not today! Sorry!

File: src/ansys/pre_commit_hooks/add_license_headers.py [hunk 1]
```diff
@@ -105,6 +108,78 @@ def set_lint_args(parser: argparse.ArgumentParser) -> argparse.Namespace:
...
+    full_path_files = []
+    for file in file_list:
+        if "win" in sys.platform:
+            split_str = file.split("/")
+            full_path_files.append(os.path.abspath(os.path.join(*split_str)))
+        else:
+            full_path_files.append(os.path.abspath(file))
+
+    return full_path_files
+
```
Comment (RobPasMue) [inline src/ansys/pre_commit_hooks/add_license_headers.py:134]: Why not use pathlib? It will handle the paths properly based on your OS

Comment (RobPasMue) [reply src/ansys/pre_commit_hooks/add_license_headers.py:134]: Also, with a pathlib.Path object you can request at any moment the absolute path. So no need to have the dedicated function to get "full" paths. Just call the absolute path method in the Path object

File: src/ansys/pre_commit_hooks/add_license_headers.py [hunk 2]
```diff
@@ -105,6 +108,78 @@ def set_lint_args(parser: argparse.ArgumentParser) -> argparse.Namespace:
...
+        changed_headers, start & end year, and git_repo
+    """
+    # Get location of LICENSE file in the repository the hook runs on
+    git_root = arg_dict["git_repo"].git.rev_parse("--show-toplevel")
+    repo_license_loc = os.path.join(git_root, "LICENSE").replace(os.sep, "/")
+    save_repo_license = shutil.copyfile(repo_license_loc, f"{repo_license_loc}_save")
+
+    # Get the location of MIT.txt in the hook's assets folder
+    hook_loc = pathlib.Path(__file__).parent.resolve()
+    hook_license_file = os.path.join(hook_loc, "assets", "LICENSES", f"{DEFAULT_LICENSE}.txt")
```
Comment (RobPasMue) [inline src/ansys/pre_commit_hooks/add_license_headers.py:153]:
```suggestion
    hook_license_file = hook_loc / "assets" / "LICENSES" / f"{DEFAULT_LICENSE}.txt"
```
Path objects can be chained like this... avoid using ``os.path`` overall

Comment (RobPasMue) [reply src/ansys/pre_commit_hooks/add_license_headers.py:153]: This applies everywhere

**Review by RobPasMue — CHANGES_REQUESTED** [review]
Looking good - but we really need to move away from os.path...

**Comment by klmcadams** [comment]
I will make a separate PR to address the os to pathlib conversions

**Pull request merged**